Client/game_sa/CPedSA.h

```diff
@@ -278,134 +308,132 @@ class CPedSA : public virtual CPed, public virtual CPhysicalSA
     CPedSAInterface*    m_pPedInterface{};
     CPedSoundSA*        m_pPedSound{};
 
-    short m_sDefaultVoiceType;
-    short m_sDefaultVoiceID;
+    std::int16_t m_sDefaultVoiceType;
+    std::int16_t m_sDefaultVoiceID;
 
     DWORD         m_dwType;
-    unsigned char m_ucOccupiedSeat;
+    std::uint8_t  m_ucOccupiedSeat;
 
 protected:
     int m_iCustomMoveAnim{ 0 };
 
 public:
-    CPedSA(CPedSAInterface* pedInterface = nullptr) noexcept;
+    CPedSA(CPedSAInterface* pedInterface = nullptr);
     ~CPedSA();
 
-    void             SetInterface(CEntitySAInterface* intInterface);
-    CPedSAInterface* GetPedInterface() { return (CPedSAInterface*)GetInterface(); }
+    void             SetInterface(CEntitySAInterface* intInterface) noexcept { m_pInterface = intInterface; }
+    CPedSAInterface* GetPedInterface() noexcept { return static_cast<CPedSAInterface*>(m_pInterface); }
     void             Init();
-    void             SetModelIndex(DWORD dwModelIndex);
+    void             SetModelIndex(DWORD modelIndex);
     void             RemoveGeometryRef();
-    void             AttachPedToEntity(DWORD dwEntityInterface, CVector* vector, unsigned short sDirection, float fRotationLimit, eWeaponType weaponType,
-                                       bool bChangeCamera);
+    void             AttachPedToEntity(DWORD entityInteface, CVector* vector, std::uint16_t direction, float rotationLimit, eWeaponType weaponType, bool changeCamera);
     void             DetachPedFromEntity();
 
     CVehicle* GetVehicle();
-    void      Respawn(CVector* position, bool bCameraCut);
-    bool      AddProjectile(eWeaponType eWeapon, CVector vecOrigin, float fForce, CVector* target, CEntity* targetEntity);
+    void      Respawn(CVector* position, bool cameraCut);
+    bool      AddProjectile(eWeaponType weaponType, CVector origin, float force, CVector* target, CEntity* targetEntity);
 
-    float GetHealth();
-    void  SetHealth(float fHealth);
+    float GetHealth() { return GetPedInterface()->fHealth; }
+    void  SetHealth(float health) { GetPedInterface()->fHealth = health; }
 
-    float GetArmor();
-    void  SetArmor(float fArmor);
+    float GetArmor() { return GetPedInterface()->fArmor; }
+    void  SetArmor(float armor) { GetPedInterface()->fArmor = armor; }
 
     float GetOxygenLevel();
-    void  SetOxygenLevel(float fOxygen);
+    void SetOxygenLevel(float oxygen);
```

Comment:
`const` for getters

Reply:
It cannot be const because GetPedInterface is not const

Reply:
If GetPedInterface is defined in the same class then you can add a const equivalent function returning that pointer

Reply:
You mean?
```cpp
    CPedSAInterface* GetPedInterface() noexcept { return static_cast<CPedSAInterface*>(m_pInterface); }
    CPedSAInterface* GetPedInterface() const noexcept { return static_cast<CPedSAInterface*>(m_pInterface); }
```

Reply:
Yes but you have to mark the return value const too in that case